webapp/src/pages/Settings.tsx

```diff
@@ -2,58 +2,623 @@ import {
+                />
+              }
+              label={metricName}
+            />
+          </Box>
+        ))}
+      </Box>
+    </Box>
+  );
+
+  const getAnalysesCustomization = () => (
+    <Box display="flex" flexDirection="column" gap={2}>
+      {ANALYSES_CUSTOMIZATION.map((customizationConfig, index) => (
+        <Box key={index} display="flex" flexDirection="column" gap={1}>
+          <Box display="flex" flexDirection="row" gap={2}>
+            {customizationConfig === "similarity"
+              ? displayToggleSectionTitle(
+                  customizationConfig,
+                  customizationConfig
+                )
+              : displaySectionTitle(customizationConfig)}
+            {Object.entries(resultingConfig?.[customizationConfig] ?? {}).map(
+              ([field, value], index) =>
+                field !== "faiss_encoder" && (
+                  <Box
+                    key={index}
+                    display="flex"
+                    flexDirection="row"
+                    justifyContent="flex-start"
+                    gap={5}
+                  >
+                    {displaySubFields(customizationConfig, field, value)}
+                  </Box>
+                )
+            )}
+          </Box>
+          {divider}
+        </Box>
+      ))}
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent="flex-start"
+        gap={5}
+      >
+        {displayToggleSectionTitle(
+          "behavioral_testing",
+          "Perturbation Testing"
+        )}
+      </Box>
+    </Box>
+  );
+
   return (
-    <Container maxWidth="sm">
-      <FormControl sx={{ width: "100%" }}>
-        <FormGroup>
-          <FormControlLabel
-            control={switchNullOrDefault("behavioral_testing")}
-            label="Perturbation testing"
-          />
-          <FormControlLabel
-            control={switchNullOrDefault("similarity")}
-            label="Similarity"
-          />
-        </FormGroup>
+    <Box sx={{ height: "100%" }}>
+      <Paper
+        variant="outlined"
+        sx={{ height: "85%", padding: 2, overflow: "auto" }}
+      >
+        <Typography variant="subtitle1">
+          View and edit certain fields from your config file. Once your changes
+          are saved, expect some delays for recomputing the affected tasks.
+        </Typography>
+        {configSection.map((section, index) => (
+          <AccordianLayout key={index} {...section}>
+            {index === 0 && getProjectConfigSection()}
+            {index === 1 && getModalContractConfigSection()}
+            {index === 2 && getAnalysesCustomization()}
+          </AccordianLayout>
+        ))}
+      </Paper>
+      <Box
+        sx={{
+          display: "flex",
+          justifyContent: "space-between",
+          alignItems: "center",
+          paddingY: 2,
+        }}
+      >
+        <Button
+          variant="contained"
+          onClick={() => setPartialConfig({ ...data })}
```

Comment:
The `partialConfig` should always only contain what's new. Here, `data` is the full config. Let's never assign it to the partial config. Instead, discarding the changes is as simple as resetting `partialConfig` to its initial value: an empty object.
```suggestion
          onClick={() => setPartialConfig({})}
```

Reply:
I understood that the reset won't work if we use the "defaultValue" attribute for the text field. But, I mentioned in the previous comment's reply that the value attribute is making the text field link "read-only". So, I am trying to analyze and fix it.

Reply:
I could find the reason behind the issue with the value attribute and pushed the fix.